File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        sortedScore = sorted(score, reverse=True)
        medals = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        rankMapping = {score: medals[i] if i < 3 else str(i + 1) for i, score in enumerate(sortedScore)}

        return [rankMapping[score] for score in score]

File: 0506-relative-ranks/0506-relative-ranks.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
var findRelativeRanks = function(score) {
    const answer = [];
    const scoreWithIndex = score.map((score, index) => [score, index]);
    
    scoreWithIndex.sort((a, b) => b[0] - a[0]);
    const rank = ["Gold Medal", "Silver Medal", "Bronze Medal"];

    for (let i = 0; i < score.length; i++) {
        const index = scoreWithIndex[i][1];
        if (i < 3) {
            answer[index] = rank[i];
        } else {
            answer[index] = String(i + 1);
        }
    }

    return answer;
};